### PR TITLE
[IL-227] Update user

### PR DESCRIPTION
**Pull request by yuliahar**
Task's name:
[IL-227](https://jira.it-sl.ru/browse/IL-227) [QA] Create API tests for updateUser functionality

Developer's full name:
Yuliya Harbacheuskaya

Changes:

Added api tests for update user endpoint.

### CODE REVIEW

File: framework/tools/generators.py [hunk 1]
```diff
@@ -27,23 +27,56 @@ def generate_string(length: int, additional_characters: list = None) -> str:
...
-    hashed_password = bcrypt.hashpw(password.encode(), bcrypt.gensalt()).decode()
-    return {
+    user_data = {
         "id": faker.uuid4(),
+        "firstName": faker.first_name()
+        if first_name_length is None
+        else generate_string(first_name_length),
+        "lastName": faker.last_name()
+        if last_name_length is None
+        else generate_string(last_name_length),
```
Comment (Sanya124) [inline framework/tools/generators.py]: Why if-else one-liners in 3 lines?))

File: framework/tools/generators.py [hunk 2]
```diff
@@ -27,23 +27,56 @@ def generate_string(length: int, additional_characters: list = None) -> str:
     return "".join(result)
 
 
-def generate_user(password: str = DEFAULT_PASSWORD) -> dict:
-    """Generating a user with the specified password
+def generate_user(
+    first_name_length: int = None,
+    last_name_length: int = None,
```
Comment (Sanya124) [inline framework/tools/generators.py]: I suggest not changing the data types. What do you think?

File: tests/user/test_user_update.py [hunk 3]
```diff
@@ -0,0 +1,320 @@
...
+@feature("Update user's information")
+class TestUpdateUser:
+    @title("Updating User Information")
+    @description(
+        "GIVEN the user is logged in, "
+        "WHEN the user sends a request to update their own information, "
+        "THEN the response code is 200 and the response body contains the current user's data."
+    )
+    def test_update_user_info(self, create_authorized_user):
+        token = create_authorized_user["token"]
```
Comment (Sanya124) [inline tests/user/test_user_update.py:24]: Can I ask you to wrap in a `step`?

File: tests/user/test_user_update.py [hunk 4]
```diff
@@ -0,0 +1,320 @@
...
+            assert_content_type(updating_user_response, "application/json")
+
+        with step("Checking the response "):
+            assert_update_user_data_matches(
+                updating_user_response.json(), user_data_to_update
+            )
+
+    fields = ["id", "email"]
+
+    @pytest.mark.parametrize("data", fields)
```
Comment (Sanya124) [inline tests/user/test_user_update.py]: Why not?
```
@pytest.mark.parametrize("data", ["id", "email"])
```

File: tests/user/test_user_update.py [hunk 5]
```diff
@@ -0,0 +1,320 @@
...
+    @title("Updating User's Email and ID is ignored")
+    @description(
+        "GIVEN the user is logged in, "
+        "WHEN the user sends a request to update their own email and id, "
+        "THEN the response code is 200 and the response body contains the current user's email and id."
+    )
+    def test_ignored_fields_on_update(self, create_authorized_user, data):
+        token = create_authorized_user["token"]
```
Comment (Sanya124) [inline tests/user/test_user_update.py:53]: In `step`?

File: tests/user/test_user_update.py [hunk 6]
```diff
@@ -0,0 +1,320 @@
...
+            assert_that(response[data], is_(create_authorized_user["user"][data]))
+
+    @title("Updating User's Own Address")
+    @description(
+        "GIVEN the user is logged in, "
+        "WHEN the user sends a request to update their own address, "
+        "THEN the response code is 200 and the response body contains the current user's address."
+    )
+    def test_update_user_address(self, create_authorized_user):
+        token = create_authorized_user["token"]
```
Comment (Sanya124) [inline tests/user/test_user_update.py:80]: In `step`?

File: tests/user/test_user_update.py [hunk 7]
```diff
@@ -0,0 +1,320 @@
...
+            )
+
+    @pytest.mark.parametrize(
+        "first_name, last_name",
+        [("John", "Doe"), ("Anne-Marie", "O'Conner"), ("O'Brien", "Smith-Jones")],
+    )
+    def test_update_user_with_valid_names(
+        self, create_authorized_user, first_name, last_name
+    ):
+        token = create_authorized_user["token"]
```
Comment (Sanya124) [inline tests/user/test_user_update.py:109]: In `step`?

File: framework/tools/generators.py (hunk 1 above)
Comment (yuliahar) [reply framework/tools/generators.py]: It's the auto-formatting feature of the Black library.

File: framework/tools/generators.py (hunk 2 above)
Comment (yuliahar) [reply framework/tools/generators.py]: Fixed.

File: tests/user/test_user_update.py (hunk 3 above)
Comment (yuliahar) [reply tests/user/test_user_update.py:24]: Thanks for the review and the idea to use an Allure step for the line token = create_authorized_user["token"]].

I think this line is just a simple setup step and not a major part of the test. Adding it as an Allure step might make our reports too detailed and less clear. What's your opinion on this?

Please also keep in mind that create_authorized_user["token"]] has already 2 allure steps

```
@title("Creating an authorized user")
@fixture(scope="function")
def create_authorized_user(postgres):
  with step("Creating user in DB"):
          user_to_create = generate_and_insert_user(postgres)
  
  with step("Authentication of user and getting token"):
          authentication_response = AuthenticateAPI().authentication(
              email=user_to_create["email"], password=user_to_create["password"]
          )
          token = authentication_response.json()["token"]
  
  yield {"user": user_to_create, "token": token}
```

File: tests/user/test_user_update.py (hunk 4 above)
Comment (yuliahar) [reply tests/user/test_user_update.py]: Fixed.

File: tests/user/test_user_update.py (hunk 5 above)
Comment (yuliahar) [reply tests/user/test_user_update.py:53]: See above please.

File: tests/user/test_user_update.py (hunk 6 above)
Comment (yuliahar) [reply tests/user/test_user_update.py:80]: See above please.

File: tests/user/test_user_update.py (hunk 7 above)
Comment (yuliahar) [reply tests/user/test_user_update.py:109]: See above please.

**Pull request merged**